src/api/spec/controllers/webui/patchinfo_controller_spec.rb

```diff
@@ -0,0 +1,269 @@
+                package: patchinfo_package.name,
+                summary: 'long enough summary is ok',
+                description: 'long enough description is also ok' * 5,
+                issueid: [769484],
+                issuetracker: ['bgo'],
+                issuesum: [],
+                issueurl: ['https://bugzilla.gnome.org/show_bug.cgi?id=769484'],
+                category: 'recommended',
+                rating: 'low',
+                packager: user.login
+  end
+
+  after do
+    Package.destroy_all
+  end
+
+  describe 'POST #new_patchinfo' do
+    before do
+      other_user
+      login user
+    end
+
+    context 'without permission to create the patchinfo package' do
+      before do
+        post :new_patchinfo, project: other_user.home_project
+      end
+
+      it { expect(response).to redirect_to(project_show_path(other_user.home_project)) }
+      it { expect(flash[:error]).to eq "No permission to create packages" }
+    end
+
+    context 'when it fails to create the patchinfo package' do
+      before do
+        Patchinfo.any_instance.stubs(:create_patchinfo).returns(false)
+        post :new_patchinfo, project: user.home_project
+      end
+
+      it { expect(response).to redirect_to(project_show_path(user.home_project)) }
+      it { expect(flash[:error]).to eq "Error creating patchinfo" }
+    end
+
+    context 'when the patchinfo package file is not found' do
+      before do
+        Package.any_instance.stubs(:patchinfo).returns(nil)
+        post :new_patchinfo, project: user.home_project
+      end
+
+      it { expect(response).to redirect_to(package_show_path(project: user.home_project, package: 'patchinfo')) }
+      it { expect(flash[:error]).to eq("Patchinfo not found for #{user.home_project.name}") }
+    end
+
+    context 'when is successfull creating the patchinfo package' do
+      before do
+        post :new_patchinfo, project: user.home_project
+      end
+
+      it { expect(response).to have_http_status(:success) }
+      it { expect(assigns(:binaries)).to be_a Array }
+      it { expect(assigns(:packager)).to eq(user.login) }
+      it { expect(assigns(:file)).not_to be_nil }
+    end
+  end
+
+  describe 'POST #updatepatchinfo' do
+    before do
+      login user
+    end
+
+    context 'without a valid patchinfo' do
+      before do
+        post :updatepatchinfo, project: user.home_project_name, package: other_package.name
+      end
+
+      it { expect(flash[:error]).to eq("Patchinfo not found for #{user.home_project_name}") }
+      it { expect(response).to redirect_to(package_show_path(project: user.home_project_name, package: other_package.name)) }
+    end
+
+    context 'with a valid patchinfo' do
+      it 'updates and redirects to edit_patchinfo' do
+        post :updatepatchinfo, project: user.home_project_name, package: patchinfo_package.name
+        expect(response).to redirect_to(action: 'edit_patchinfo', project: user.home_project_name, package: patchinfo_package.name)
```

Comment:
Ok, I added the expects+with and the unit test will be added to our trello board :+1: